desktop-exporter/otlp_payload_processor_test.go

```diff
@@ -0,0 +1,57 @@
+package desktopexporter
+
+import (
+	"context"
+	"encoding/hex"
+	"testing"
+	"time"
+
+	"github.com/CtrlSpice/desktop-collector/desktop-exporter/testdata"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	resourceCount = 2
+	scopeCount    = 3
+	spanCount     = 4
+)
+
+func TestExtractSpans(t *testing.T) {
+	traces := testdata.GenerateTraces(resourceCount, scopeCount, spanCount)
+	spans := extractSpans(context.Background(), traces)
```

Comment:
an assertion against `spans.length` would be a good addition I think

Reply:
Good call!